### PR TITLE
Leaflet integration

### PR DESCRIPTION
**Pull request by artturimatias**
- clustering of markers (experiment)
- map layers (users can see several overlapping maps)
- geosearch

### CODE REVIEW

File: js/ui.js [hunk 1]
```diff
@@ -1,16 +1,169 @@
...
-  } else {
-    $('.dropdown').slideDown();
-    $('#first-dropdown-link').focus();
-  }
+
+$(document).ready(function(){
+	$('#tabs').tabs({'active':1});
+	$('#tabs' ).tabs('disable', 0 );
+	$('#tabs' ).tabs('disable', 3 );
+    $('.alert-container').hide();
```
Comment (Abbe98) [inline js/ui.js:7]: Why would we do this? 

No errors is visible or accessible...

File: js/app.js [hunk 2]
```diff
@@ -1,32 +1,221 @@
...
-      console.log('invalid alert type');
-    }
-  }
-};
+// Leaflet integration for Map Warper (Wikimaps)
+
+
+var W_LEAFLET = {
+    map_div:		'map',
+    base_url:		'http://warper.wmflabs.org',
```
Comment (Abbe98) [inline js/app.js:6]: This is defined in `W_API` why duplicate it?

File: js/app.js [hunk 3]
```diff
@@ -1,32 +1,221 @@
...
+    bounds_weight: 	2,
+    markers: 		[],
+	init: function () {
+		
+		// set up the map
+		map = new L.Map(this.map_div);
+		this.map = map;
+
+		// create the base map layer 
+		var osmUrl='http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png';
```
Comment (Abbe98) [inline js/app.js]: "layerUrl"?

File: js/ui.js [hunk 4]
```diff
@@ -1,16 +1,169 @@
...
-    $('.dropdown').slideUp();
-  } else {
-    $('.dropdown').slideDown();
-    $('#first-dropdown-link').focus();
-  }
+
+$(document).ready(function(){
+	$('#tabs').tabs({'active':1});
+	$('#tabs' ).tabs('disable', 0 );
+	$('#tabs' ).tabs('disable', 3 );
```
Comment (Abbe98) [inline js/ui.js:6]: Tabs are now broken...

File: js/app.js [hunk 5]
```diff
@@ -1,32 +1,221 @@
...
+		if (this.clustering) this.map.addLayer(this.clusterLayer);
+		this.map.addLayer(this.featureLayer);
+		if (this.bounds) this.map.addLayer(this.boundsLayer);
+
+		this.map.on('popupclose', function() {
+			W_LEAFLET.map.removeLayer(W_LEAFLET.boundsMarker);
+		});		
+	},
+
+	setView: function (lat, lon, zoom) {
```
Comment (Abbe98) [inline js/app.js:48]: Does not Leaflet provide a setView or equal method?

**Comment by Abbe98** [comment]
This seams to use both CamelCase and some underscore stuff...

Overall formatting seams to be broken(JS/CSS)...

**Comment by Abbe98** [comment]
The used proxy does not support CORS?

File: index.html [hunk 6]
```diff
@@ -157,18 +101,20 @@
...
 			</div>
 			<div class="alert-container">
 			</div>
 		</div>
 		<!-- RIGHT column ends -->
 	</div>
+	<script src="leaflet/leaflet.js"></script>
+	<script src="leaflet/leaflet.markercluster.js"></script>
 	<script src="js/ui.js"></script>
+	<script src="js/api.js"></script>
```
Comment (Abbe98) [inline index.html]: Duplicate request?

File: css/wikimaps-style.css [hunk 7]
```diff
@@ -344,3 +344,143 @@ nav {
...
+	position: relative;
+	text-align: left;
+}
+
+#maps .ui-slider-handle {
+    background-color:#85A3C2;
+}
+
+
+.ui-slider .ui-slider-handle {
```
Comment (Abbe98) [inline css/wikimaps-style.css:418]: copy paste stuff could be cleaned up

File: css/wikimaps-style.css [hunk 8]
```diff
@@ -344,3 +344,143 @@ nav {
...
+	float:right;
+    left:20px;
+	border: #ccc 1px solid;
+	padding:2px;
+	width:6px;
+	height:76px;
+	background:#ccc;
+}
+
+#layer_list > div {
```
Comment (Abbe98) [inline css/wikimaps-style.css]: Could we reuse existing CSS?

**Comment by Abbe98** [comment]
This currently throws multiply offline errors.

Store API availability in a variable and check if it has changed?

**Comment by Abbe98** [comment]
`W_LEAFLET.setView()` is called twice at startup.

File: js/app.js (hunk 2 above)
Comment (artturimatias) [inline js/app.js:6]: You mean 'http://warper.wmflabs.org'? The other is for warper api and other is for map tiles. This allows temporarily use proxy for api.

File: js/ui.js (hunk 4 above)
Comment (artturimatias) [inline js/ui.js:6]: Not broken, just disables the ones that are not used.

File: js/app.js (hunk 5 above)
Comment (artturimatias) [inline js/app.js:48]: It is just a wrapper, so that ui code does not have to deal directly with Leaflet.

File: js/ui.js (hunk 1 above)
Comment (artturimatias) [inline js/ui.js:7]: Sorry, it was just a quick fix for alert-container blocking Leaflet events. I'll fix it.

File: css/wikimaps-style.css (hunk 7 above)
Comment (artturimatias) [inline css/wikimaps-style.css:418]: Sorry and thanks for the fix.

File: js/ui.js (hunk 4 above)
Comment (Abbe98) [inline js/ui.js:6]: Why would we do that during development?

File: js/app.js (hunk 2 above)
Comment (Abbe98) [inline js/app.js:6]: Makes sense but is they the same?

Comment (artturimatias) [inline js/app.js:6]: They are the same but I'm using currently proxy for testing: http://wikihacks.opendimension.org/wmaps_html/ so we can have a working demo.

File: js/ui.js (hunk 4 above)
Comment (artturimatias) [inline js/ui.js:6]: Yes, I'll remove those.

File: css/wikimaps-style.css (hunk 7 above)
Comment (artturimatias) [inline css/wikimaps-style.css:418]: Proxy is just a quick hack so we cun run a public demo at least somewhere.

**Pull request merged**